tests/mocha/field_colour_test.js

```diff
@@ -0,0 +1,179 @@
+    //  Here I have it converting to lowercase, because I think that looks
+    //  better when the block collapses.
+    test('#FFFFFF', function() {
+      var colourField = new Blockly.FieldColour('#FFFFFF');
+      assertValue(colourField, '#ffffff', '#fff');
+    });
+    test('#ffffff', function() {
+      var colourField = new Blockly.FieldColour('#ffffff');
+      assertValue(colourField, '#ffffff', '#fff');
+    });
+    test('#FFFF00', function() {
+      var colourField = new Blockly.FieldColour('#FFFF00');
+      assertValue(colourField, '#ffff00', '#ff0');
+    });
+    test('#ffff00', function() {
+      var colourField = new Blockly.FieldColour('#ffff00');
+      assertValue(colourField, '#ffff00', '#ff0');
+    });
+    test('#BCBCBC', function() {
+      var colourField = new Blockly.FieldColour('#BCBCBC');
+      assertValue(colourField, '#bcbcbc', '#bcbcbc');
+    });
+    test('#bcbcbc', function() {
+      var colourField = new Blockly.FieldColour('#bcbcbc');
+      assertValue(colourField, '#bcbcbc', '#bcbcbc');
+    });
+  });
+  suite('fromJson', function() {
+    test('Empty', function() {
+      var colourField = new Blockly.FieldColour.fromJson({});
+      assertValueDefault(colourField);
+    });
+    test('Null', function() {
+      var colourField = new Blockly.FieldColour.fromJson({ colour:null });
+      assertValueDefault(colourField);
+    });
+    test('Undefined', function() {
+      var colourField = new Blockly.FieldColour.fromJson({ colour:undefined });
+      assertValueDefault(colourField);
+    });
+    test('Non-Parsable String', function() {
+      var colourField = new Blockly.FieldColour.fromJson({ colour:'bad' });
+      assertValueDefault(colourField);
+    });
+    test('#FFFFFF', function() {
+      var colourField = Blockly.FieldColour.fromJson({ colour: '#FFFFFF' });
+      assertValue(colourField, '#ffffff', '#fff');
+    });
+    test('#ffffff', function() {
+      var colourField = Blockly.FieldColour.fromJson({ colour: '#ffffff' });
+      assertValue(colourField, '#ffffff', '#fff');
+    });
+    test('#FFFF00', function() {
+      var colourField = Blockly.FieldColour.fromJson({ colour: '#FFFF00' });
+      assertValue(colourField, '#ffff00', '#ff0');
+    });
+    test('#ffff00', function() {
+      var colourField = Blockly.FieldColour.fromJson({ colour: '#ffff00' });
+      assertValue(colourField, '#ffff00', '#ff0');
+    });
+    test('#BCBCBC', function() {
+      var colourField = Blockly.FieldColour.fromJson({ colour: '#BCBCBC' });
+      assertValue(colourField, '#bcbcbc', '#bcbcbc');
+    });
+    test('#bcbcbc', function() {
+      var colourField = Blockly.FieldColour.fromJson({ colour: '#bcbcbc' });
+      assertValue(colourField, '#bcbcbc', '#bcbcbc');
+    });
+  });
+  suite('setValue', function() {
+    test('#ffffff -> Null', function() {
+      var colourField = new Blockly.FieldColour('#ffffff');
```

Comment:
Creating the colourField can be moved into setup for this suite. Also, you should use a value other than the default so you can distinguish when it's falling back to the default vs the current value.